crates/viewer/re_space_view_spatial/src/visualizers/videos.rs

```diff
@@ -0,0 +1,236 @@
+                re_log::error_once!("Failed to create rectangle draw data from images: {err}");
+            }
+        }
+
+        Ok(draw_data_list)
+    }
+
+    fn data(&self) -> Option<&dyn std::any::Any> {
+        Some(self.data.as_any())
+    }
+
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn as_fallback_provider(&self) -> &dyn re_viewer_context::ComponentFallbackProvider {
+        self
+    }
+}
+
+// NOTE: Do not put profile scopes in these methods. They are called for all entities and all
+// timestamps within a time range -- it's _a lot_.
+impl AssetVideoVisualizer {
+    // TODO(jan): need to write to `self.data` at some point
+    #[allow(clippy::unused_self)]
+    fn process_data(
+        &mut self,
+        ctx: &QueryContext<'_>,
+        render_ctx: &RenderContext,
+        rectangles: &mut Vec<TexturedRect>,
+        ent_context: &SpatialSceneEntityContext<'_>,
+        data: impl Iterator<Item = AssetVideoComponentData>,
+        current_time_seconds: f64,
+        query_result_hash: Hash64,
+    ) {
+        let entity_path = ctx.target_entity_path;
+
+        for data in data {
+            let timestamp_s = current_time_seconds - data.index.time.as_f64() / 1e9;
+            let video = AssetVideo {
+                blob: data.blob.clone().into(),
+                media_type: data.media_type.clone().map(Into::into),
+            };
+
+            let primary_row_id = data.index.row_id;
+            let picking_instance_hash = re_entity_db::InstancePathHash::entity_all(entity_path);
+
+            let video = ctx.viewer_ctx.cache.entry(|c: &mut VideoCache| {
+                c.entry(
+                    &entity_path.to_string(),
+                    VideoCacheKey {
+                        versioned_instance_path_hash: picking_instance_hash
+                            .versioned(primary_row_id),
+                        query_result_hash,
+                        media_type: data.media_type.clone().map(Into::into),
+                    },
+                    &video.blob,
+                    video.media_type.as_ref().map(|v| v.as_str()),
+                    render_ctx,
+                )
+            });
+
+            if let Some(video) = video {
+                let mut video = video.lock();
+                let texture = video.get_frame(timestamp_s);
+
+                let world_from_entity = ent_context
+                    .transform_info
+                    .single_entity_transform_required(ctx.target_entity_path, "Video");
+                // TODO(jan): texture extents are wrong, this only shows roughly the top-left corner of the video
+                let textured_rect = TexturedRect {
+                    top_left_corner_position: world_from_entity.transform_point3(Vec3::ZERO),
+                    extent_u: world_from_entity.transform_vector3(Vec3::X * video.width() as f32),
+                    extent_v: world_from_entity.transform_vector3(Vec3::Y * video.height() as f32),
```

Comment:
I think there are videos with non-square pixels, which means there is probably an aspect ratio to take into account here somewhere (and for the bounding box below)